### ISSUE TITLE
Jax.vjp() call not working with PhysicsData object 

### ISSUE THREAD

**Issue by rgjini** [climate-analytics-lab/jax-gcm#80]
The issue can be reproduced in the [autodiff_vert_diffusion.ipynb](https://github.com/climate-analytics-lab/jax-gcm/blob/autodiff-test/notebooks/autodiff_vert_diffusion.ipynb) notebook under the notebooks tab, located on the [autodiff-test branch](https://github.com/climate-analytics-lab/jax-gcm/tree/autodiff-test).  

We are trying to use [jax.vjp ](https://github.com/climate-analytics-lab/jax-gcm/blob/c9e81606db37719b2062ee9d4e87f3112cf1df19/notebooks/autodiff_vert_diffusion.ipynb#L750) to take the gradient of the [get_vertical_diffusion_tend](https://github.com/climate-analytics-lab/jax-gcm/blob/c9e81606db37719b2062ee9d4e87f3112cf1df19/jcm/vertical_diffusion.py#L16) function within the vertical_diffusion.py script.  The function takes in two parameters, a PhysicsState object and a PhysicsData object (located in the [physics_data.py](https://github.com/climate-analytics-lab/jax-gcm/blob/autodiff-test/jcm/physics_data.py) script).  When we call vjp, we get a "Shapes must be 1D sequences of concrete values of integer type" error.  

We believe the issue comes from when jax tries to take the gradient of the [nodal_shape parameter in the PhysicsData constructor](https://github.com/climate-analytics-lab/jax-gcm/blob/c9e81606db37719b2062ee9d4e87f3112cf1df19/jcm/physics_data.py#L283).  We've tried printing out the shape, and when the actual PhysicsData object is constructed, it returns the correct shape.  But when vjp is called, the shape returned is now a Traceable Array (not a tuple of integers).  We think we need to figure out a way for jax to ignore the nodal_shape parameter within the data class object, but we're unsure if that will fix the problem.
